### PR TITLE
Misc Backend Cleanup 1

### LINKED ISSUE #461: Misc Backend Cleanup 1
While working on [Tooling issue 453](https://github.com/isomerpages/isomercms-backend/issues/453) we decided the that the following changes would make the isomercms-backend code easier to navigate.

- [x] Move all isomercms-backend code into a src subdir
- [x] Merge `newroutes` and `routes`, `newmiddleware` and `middlware` or convert to a numeric versioning scheme.
- [x] Move `MailClient`, `SmsClient`, and `TotpGenerator` into `utilServices`

Moved other issues to #469

### PR DESCRIPTION
**Pull request by rc-davis**
This was rebased off of `fix/git-cors`, so it will bring in #471 as well. We should rebase off of `develop` if we merge #471. 

This PR will address some of the isomercms-backend refactoring for Q3 2022 tasks, as listed in #461. The changes in this PR are far-reaching and likely to conflict with many PRs, so it would be good to merge this into develop before we make any long-running branches.

This is a big PR, so let me focus your attention on what's important:

- Most changes simply move files into a new directory. You can review these simply by looking at the structure of the files in the following directories:
  - src (top level directories only)
  - src/routes
    - old routes have moved into v1
    - newroutes have moved into v2
  - src/middleware 
    - newmiddleware merged into here
    - the obsolete middleware/auth.js has been deleted, and newmiddleware/auth.js has moved into its place.
  - src/services/utilServices
    - MailClient, ~~SmsClient, and TotpGenerator~~ moved here from src/services/identity
- The following files might be worth a closer look, as they wire up tools that need to know where source files live.
  - .eslintrc.json
  - .sequelizerc
  - jest.config.js
  - package.json
  - tsconfig.json
  - tsconfig.build.json

closes #461

### CODE REVIEW

File: src/integration/Users.spec.ts [hunk 1]
```diff
@@ -76,7 +76,7 @@ describe("Users Router", () => {
       expect(actual.statusCode).toBe(expected)
     })
 
-    it("should return 400 when there is no email in the request body", async () => {
+    it("should return 200 when email sending is successful", async () => {
```
[the remark this answers is not in the document]
Comment (rc-davis) [reply src/integration/Users.spec.ts:79]: Thanks @seaerchin! I thought I fixed that!

File: src/middleware/index.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import FormSG from "@opengovsg/formsg-sdk"
 import express, { RequestHandler } from "express"
 
-import FormsProcessingService from "@root/services/middlewareServices/FormsProcessingService"
 import AuthService from "@services/identity/AuthService"
 import { AuthMiddlewareService } from "@services/middlewareServices/AuthMiddlewareService"
+import FormsProcessingService from "@services/middlewareServices/FormsProcessingService"
 
 import { AuthMiddleware } from "./auth"
```
[the remark this answers is not in the document]
Comment (rc-davis) [reply src/middleware/index.ts:9]: Thanks @seaerchin. Normally I would say yes, but `index.ts` wasn't used to re-export definitions from the files contained in `newmiddleware`. Instead, it provides functions that simplify injection of service dependencies. There weren't any similar functions for the old middleware, and now didn't seem like the time to add them. (I'm trying to minimize the changes in this PR).

File: src/services/identity/UsersService.ts [hunk 3]
```diff
@@ -3,10 +3,9 @@ import { Sequelize } from "sequelize-typescript"
 import { RequireAtLeastOne } from "type-fest"
 
 import { User, Whitelist } from "@database/models"
-
-import MailClient from "./MailClient"
-import SmsClient from "./SmsClient"
-import TotpGenerator from "./TotpGenerator"
+import MailClient from "@services/utilServices/MailClient"
+import SmsClient from "@services/utilServices/SmsClient"
+import TotpGenerator from "@services/utilServices/TotpGenerator"
```
[the remark this answers is not in the document]
Comment (rc-davis) [reply src/services/identity/UsersService.ts:8]: I'm happy to shift this back. Does anyone else have an opinion?

File: src/middleware/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (rc-davis) [reply src/middleware/index.ts:9]: @seaerchin could you be more specific about what you had in mind? Maybe we could make an issue for this and tackle it later.

File: src/services/identity/__tests__/UsersService.spec.ts [hunk 4]
```diff
@@ -2,10 +2,10 @@ import { Sequelize } from "sequelize-typescript"
 import { ModelStatic } from "sequelize/types"
 
 import { User, Whitelist } from "@root/database/models"
+import MailClient from "@services/utilServices/MailClient"
+import SmsClient from "@services/utilServices/SmsClient"
+import TotpGenerator from "@services/utilServices/TotpGenerator"
```
[the remark this answers is not in the document]
Comment (rc-davis) [reply src/services/identity/__tests__/UsersService.spec.ts]: Moved back.

File: src/services/identity/UsersService.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (rc-davis) [reply src/services/identity/UsersService.ts]: Moved back.

File: src/middleware/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (rc-davis) [reply src/middleware/index.ts:9]: I added an item for reexports to #469.

**Review by seaerchin — APPROVED** [review]
lgtm

**Review by prestonlimlianjie — APPROVED** [review]
lgtm

**Pull request merged**